kong/router/compat.lua

```diff
@@ -188,7 +190,8 @@ local function get_expression(route)
     end
 
     if src_gen or dst_gen then
-      return expr_buf:get()
+      local str = expr_buf:get()
+      return str
```

Comment:
``` suggestion
      -- returns a local variable instead uses a tail call
      -- to avoid NYI
      local str = expr_buf:get()
      return str
```